maxim/apis/maxim_apis.py

```
@@ -689,45 +696,205 @@ def get_folders(self) -> List[Folder]:
-        Add entries to a dataset.
 
+        # Process each attachment in the entry
+        file_attachments = []
+        for attachment in entry.payload:
+            try:
+                file_data, mime_type, size = self._process_attachment(attachment)
+                
+                # Validate MIME type
+                if not mime_type or mime_type == 'application/octet-stream':
+                    # Try to infer MIME type from file extension if available
+                    if hasattr(attachment, 'name') and attachment.name:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)